src/Components/Browser.JS/src/Boot.Server.ts

```diff
@@ -6,70 +6,97 @@ import { OutOfProcessRenderBatch } from './Rendering/RenderBatch/OutOfProcessRen
 import { internalFunctions as uriHelperFunctions } from './Services/UriHelper';
 import { renderBatch } from './Rendering/Renderer';
 import { fetchBootConfigAsync, loadEmbeddedResourcesAsync } from './BootCommon';
+import { CircuitHandler } from './Platform/Circuits/CircuitHandler';
 
-let connection : signalR.HubConnection;
+async function boot(reconnect: boolean = false) {
+  const circuitHandlers = new Array<CircuitHandler>();
+  window['Blazor'].addCircuitHandler = (circuitHandler: CircuitHandler) => circuitHandlers.push(circuitHandler);
 
-function boot() {
-  // In the background, start loading the boot config and any embedded resources
-  const embeddedResourcesPromise = fetchBootConfigAsync().then(bootConfig => {
-    return loadEmbeddedResourcesAsync(bootConfig);
-  });
+  await startCicuit();
```

Comment:
There's a typo in `startCicuit`, but more importantly, can we avoid the nested function? As far as I can tell we don't need `startCicuit` at all - it doesn't get called from anywhere except here, so we could just inline it.